code/game/objects/items/storage/boxes.dm

```diff
@@ -476,45 +405,85 @@
 		isopened = 1
 		icon_state = "mealpackopened"
 
-//Fillable Ammo Box
-/obj/item/storage/box/ammo
-	name = "\improper Ammo Box"
-	desc = "A large ammo box that can be filled with almost any magazine type. Compact and deployable for moving ammo to the front lines."
-	icon_state = "ammobox"
+/** Fillable box
+ *
+ * Deployable box with fancy visuals of its contents
+ * Visual content defined in the icon_state_mini var in /obj/item
+ * All other visuals that do not have a icon_state_mini defined are in var/assoc_overlay
+ */
+/obj/item/storage/box/visual
+	name = "generic box"
+	desc = "This box is able to hold a wide variety of supplies."
+	icon = 'icons/obj/items/storage/storage_boxes.dmi'
+	icon_state = "mag_box"
+	item_state = "mag_box"
+	w_class = WEIGHT_CLASS_HUGE
+	slowdown = 0.4 // Big unhandly box
 	max_w_class = 4
-	storage_slots = 30
-	max_storage_space = 60	//SMG and pistol sized (tiny and small) mags can fit all 30 slots, normal (LMG and AR) fit 20
+	storage_slots = 32 // 8 images x 4 items
+	max_storage_space = 64
 	use_to_pickup = TRUE
 	can_hold = list(
-		/obj/item/ammo_magazine/pistol,
-		/obj/item/ammo_magazine/smg,
-		/obj/item/ammo_magazine/rifle,
-		/obj/item/ammo_magazine/magnum,
-		/obj/item/ammo_magazine/revolver,
-		/obj/item/ammo_magazine/acp,
-		/obj/item/ammo_magazine/standard_lmg,
-		/obj/item/ammo_magazine/standard_smartmachinegun,
-		/obj/item/ammo_magazine/m412l1_hpr,
-		/obj/item/ammo_magazine/shotgun,
-		/obj/item/ammo_magazine/sniper,
+		/obj/item, //This box should normally be unobtainable so here we go
 	)
-
+	///Assoc list linking each item to it's corresponding icon_state for the overlays if the atom does not have a item_state_mini var.
```

Comment:
how does it happen that the var is not defined? im confused a tad here

Reply:
woops, forgot to delete that. Used it as an override so other non magazine items could have an assigned icon. Since i moved the icon_state_mini var to /obj/item instead its not needed. Deleted.